py-polars/polars/internals/lazy_functions.py

```diff
@@ -2044,3 +2047,117 @@ def coalesce(
     """
     exprs = pli.selection_to_pyexpr_list(exprs)
     return pli.wrap_expr(_coalesce_exprs(exprs))
+
+
+@overload
+def from_epoch(
+    column: str | pli.Expr | pli.Series,
+    unit: EpochTimeUnit = ...,
+    *,
+    eager: Literal[False],
+) -> pli.Expr:
+    ...
+
+
+@overload
+def from_epoch(
+    column: str | pli.Expr | pli.Series | Sequence[int],
+    unit: EpochTimeUnit = ...,
+    *,
+    eager: Literal[True],
+) -> pli.Series:
+    ...
+
+
+@overload
+def from_epoch(
+    column: pli.Series | Sequence[int],
+    unit: EpochTimeUnit = ...,
+    *,
+    eager: Literal[True] = ...,
+) -> pli.Series:
+    ...
+
+
+@overload
+def from_epoch(
+    column: str | pli.Expr,
+    unit: EpochTimeUnit = ...,
+    *,
+    eager: Literal[False] = ...,
+) -> pli.Expr:
+    ...
+
+
+@overload
+def from_epoch(
+    column: str | pli.Expr | pli.Series | Sequence[int],
+    unit: EpochTimeUnit = ...,
+    *,
+    eager: bool = ...,
+) -> pli.Expr | pli.Series:
+    ...
+
+
+def from_epoch(
+    column: str | pli.Expr | pli.Series | Sequence[int],
+    unit: EpochTimeUnit = "s",
+    *,
+    eager: bool = False,
+) -> pli.Expr | pli.Series:
+    """
+    Utility function that parses an epoch timestamp (or Unix time) to Polars Datetime.
+
+    If possible, the unit will persist in `pl.Datetime`, defaults to "ns".
+
+    Parameters
+    ----------
+    column
+        Series or expression to parse integers to pl.Datetime.
+    unit
+        The unit of the timesteps since epoch time.
+    eager
+        If eager evaluation is `True`, a Series is returned instead of an Expr.
+
+    Examples
+    --------
+    >>> df = pl.DataFrame({"timestamp": [1666683077, 1666683099]}).lazy()
+    >>> df.select(pl.from_epoch(pl.col("timestamp"), unit="s")).collect()
+    shape: (2, 1)
+    ┌─────────────────────┐
+    │ timestamp           │
+    │ ---                 │
+    │ datetime[μs]        │
+    ╞═════════════════════╡
+    │ 2022-10-25 07:31:17 │
+    ├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+    │ 2022-10-25 07:31:39 │
+    └─────────────────────┘
+
+    """
+    if isinstance(column, str):
+        column = col(column)
+    elif not isinstance(column, (pli.Series, pli.Expr)):
+        column = pli.Series(column)  # Sequence input handled by Series constructor
+
+    if unit == "d":
```

Comment:
So what is the expected dtype of this function? `Datetime` for any unit below days?

Reply:
for `"d"` it is the number of days since epoch (which is apparently a thing as well), therefore expected output is a series with dtype `pl.Date`.

Reply:
I think we should document the output dtypes. So that date for "d" and datetime for the rest I think.

Reply:
Can also cast the date to datetime for more consistency, although the pl.Date precision of the days since epoch timestamp makes more sense

Reply:
Yes, I think the date is fine. Just that it's clear which dtype we can expect.